web/src/routes/MyPages/AuthListPage.tsx

```diff
@@ -1,5 +1,279 @@
+  }
+`;
+
+const Form = styled.form`
+  display: flex;
+  flex-direction: column;
+`;
+
+const AuthListTitle = styled.span`
+  text-align: left;
+  font-size: calc(2vh + 2vmin);
+  font-weight: 900;
+  margin-bottom: 2vh;
+  color: ${(props) => props.theme.colors.default};
+  cursor: pointer;
+`;
+
+const BeforeText = styled.span`
+  text-align: left;
+  font-size: calc(1vh + 1.4vmin);
+  font-weight: 900;
+  min-width: 80px;
+  max-width: 40vw;
+  width: fit-content;
+  color: ${(props) => props.theme.colors.default};
+`;
+
+const AfterText = styled.span`
+  text-align: left;
+  font-size: calc(1vh + 1.4vmin);
+  font-weight: 900;
+  min-width: 80px;
+  max-width: 40vw;
+  width: fit-content;
+  color: ${(props) => props.theme.colors.btnhover};
+`;
+
+type State = {
+  kakao: boolean;
+  naver: boolean;
+  google: boolean;
+  identityCard: boolean;
+  driverCard: boolean;
+  schoolMail: boolean;
+};
+
+type Action =
+  | { type: 'AUTH_KAKAO'; payload: boolean }
+  | { type: 'AUTH_NAVER'; payload: boolean }
+  | { type: 'AUTH_GOOGLE'; payload: boolean }
+  | { type: 'AUTH_IDENTITY'; payload: boolean }
+  | { type: 'AUTH_DRIVER'; payload: boolean }
+  | { type: 'AUTH_SCHOOLMAIL'; payload: boolean };
+
+// test
+const initialState = {
+  kakao: true,
+  naver: false,
+  google: true,
+  identityCard: false,
+  driverCard: false,
+  schoolMail: false,
+};
+
+const reducer = (state: State, action: Action): State => {
+  switch (action.type) {
+    case 'AUTH_KAKAO':
+      return { ...state, kakao: action.payload };
+    case 'AUTH_NAVER':
+      return { ...state, naver: action.payload };
+    case 'AUTH_GOOGLE':
+      return { ...state, google: action.payload };
+    case 'AUTH_IDENTITY':
+      return { ...state, identityCard: action.payload };
+    case 'AUTH_DRIVER':
+      return { ...state, driverCard: action.payload };
+    case 'AUTH_SCHOOLMAIL':
+      return { ...state, schoolMail: action.payload };
+    default:
+      return state;
+  }
+};
+
+export default function MyPage() {
+  const [theme, toggleTheme] = useTheme();
+  const navigate = useNavigate();
+
+  const [state, dispatch] = useReducer(reducer, initialState);
+
+  // Input Data list
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+  };
+
+  return (
+    <Container theme={theme}>
+      <Header theme={theme} toggleTheme={toggleTheme} />
+      <AuthListContainer theme={theme}>
+        <Form onSubmit={handleSubmit}>
+          <AuthListTitle theme={theme} style={{ marginBottom: '35px' }} onClick={() => navigate('../mypage')}>
+            마이페이지 &gt; 인증정보 목록{' '}
+          </AuthListTitle>
+          <ContainerBox theme={theme} onClick={() => navigate('../mypage/auth-list')} disabled={state.kakao}>
+            <KakaoTalk type="submit" />
+            {!state.kakao ? (
+              <BeforeText theme={theme}>카카오 본인인증 바로가기</BeforeText>
+            ) : (
+              <AfterText theme={theme}>카카오 본인인증 완료</AfterText>
```

Comment:
As  `<BeforeText>` and `<AfterText>` only differs in its child text, please just make one component and use lambda function. 

It'll look like `<YourTextComponent theme{theme}>{!state.kakao ? "카카오 본인인증 바로가기" : "카카오 본인인증 완료"}</YourTextComponent>`.

But for another components for `kakao`, `naver`, `google`, `id`, ... they're all **duplicated** way too much. Please combine them in one function with switch case or something else so that it looks more neat. You need to avoid duplicated logic.

One step more, following patterns are duplicated on this page. 
```typescript
<ContainerBox>
  <AuthLogo>
    <YourTextComponent theme{theme}>
      {!state.AUTH_NAME ? `${AUTH_NAME} 본인인증 바로가기` : `${AUTH_NAME} 본인인증 완료`}
    </YourTextComponent>
    <CheckOrArrowImage />
  </AuthLogo>
</ContainerBox>
```

But this is bad practice as mentioned above so you need to make separate component for this.